server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -253,6 +257,12 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
      */
     public static final String VIRTUAL_DATA_BLOB_PREFIX = "v__";
 
+    public static final String SNAPSHOT_REPOSITORY_DATA_CACHE_SIZE_SETTING_NAME = "snapshot.repository_data.cache.size";
```

Comment:
I think there was quite a bit work done to centralize cache management:
 - we have `CacheService` to create cache (and basically provide an opportunity to make sure the caches configurations are sane)
 - we have `NodeCacheStats` to provide the cache utilization statistics

I looks to me this is the direction we should be taking, no? @peteralfonsi @sgup432 what do you think folks?

Reply:
Result caching is substantially more complex than what is happening here. I would really really like to avoid exposing anything in a stat API about this behavior. I would also like to avoid adding a cluster setting too. The original proposal was "that we have cache size which is x% of heap size" instead of just a fixed 500KB threshold. The use case here is to simply memoize the latest snapshot metadata because there's a good chance it will need to be accessed again. I'm wondering if changing the threshold to a fixed percentage of the heap (I know settling on the specific value can be difficult) plus the new soft reference behavior as a backstop against OOMs is good enough? No need for any setting or stat or anything like that. @ashking94 @reta what do you think?

Reply:
>  The original proposal was "that we have cache size which is x% of heap size" instead of just a fixed 500KB threshold. 

To me this is the problem: what is relationship between heap size and snapshot metadata? Fixed size is at least predictable measure, but 1% of 32Gb heap is ~300Mb. I think just keeping the weak / soft reference to the snapshot metadata should be sufficient, if JVM is low on heap, those will be reclaimed.

Reply:
Hello, @reta!
The snapshot metadata grows with the number of indices and snapshots. As a best practice, the number of indices should be limited based on the available heap memory. I’ve provided more details in [my comment](https://github.com/opensearch-project/OpenSearch/issues/16298#issuecomment-2444951524), including an estimate of how much the snapshot metadata cache size may increase according to AWS-recommended best practices.
Thank you for your review!

Reply:
> The snapshot metadata grows with the number of indices and snapshots.

Yes, but to be clear, how often do we access _all_ the snapshots?  In my experience we are mostly concerned with the most recent one(s) (for restoring) or the oldest one(s) (which contain the data about indices that haven't changed much).  

> As a best practice, the number of indices should be limited based on the available heap memory.

This alone is a good argument for making the value a fixed percentage of heap, but I'm not sure that multiplying by the number of snapshots is necessary beyond a day or three's worth.

Reply:
> Yes, but to be clear, how often do we access all the snapshots?

@dbwiddis The metadata is a single monolithic blob. We have to fetch the whole blob and deserialize it in order to get information about any single snapshot. In this case we are only ever "caching" a single entry, which is the last version of the metadata blob we fetched from remote.